pagerduty/resource_pagerduty_user.go

```diff
@@ -105,11 +107,24 @@ func resourcePagerDutyUser() *schema.Resource {
 				Optional: true,
 				Default:  "Managed by Terraform",
 			},
+
+			"license": {
```

Comment:
To avoid making use of this workaround what We've been doing is to set the attribute type as `string` and in its implementation We internally handle the reference type, like in [`pagerduty_service.escalation_policy`](https://github.com/PagerDuty/terraform-provider-pagerduty/blob/1035d95106bd4db130fefff2cb48e505ac098a2d/pagerduty/resource_pagerduty_service.go#L170) and many others We have rolled out.

Reply:
Since this attribute's value is an object in the API, I think it should stay that way for terraform. I think it get's confusing to change types. What we can do is allow the license to be set to `data.pagerduty_license.<name>` to make it easier to manage user licenses. I will push that change and see what you think about it.

Reply:
I've changed the schema to `TypeString`